### ISSUE TITLE
Script running on GitHub makes the page download on Microsoft Edge

### ISSUE THREAD

**Issue by ccloli** [Tampermonkey/tampermonkey#315]
When a script is running on GitHub, the page would be force downloaded on Microsoft Edge.

![Screenshot](https://cloud.githubusercontent.com/assets/8115912/18208090/8b181b8c-7160-11e6-9276-1a4a67a5f1fb.png)
_(Sorry this screenshot is in Chinese. It said "What do you want to do with ccloli.html? From: github.com", and the three buttons are "Save", "Save As", "Cancel")_

Example script, then visit any page that host is github.com : 
```js
// ==UserScript==
// @name        I can do nothing but download GitHub
// @include     *
// ==/UserScript==
```
Or with a specific page, then visit https://github.com/ccloli :
```js
// ==UserScript==
// @name        I can do nothing but download GitHub
// @include     https://github.com/ccloli
// ==/UserScript==
```

It do nothing on any other browsers, but only download pages on Edge.

I'm not sure if this will happen on other pages, at this time and AFAIK it only happens on GitHub.

User Agent: `Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KMTML, like Gecko) Chrome/51.0.2704.79 Safari/537.36 Edge/14.14393`
Tampermonkey Version: `v4.2.5854`

**Comment by lychichem** [comment]
I can always reproduce this issue with this script (https://greasyfork.org/zh-CN/scripts/2998-search-by-image/code)

**Comment by derjanb** [comment]
As a possible workaround please execute this at the background page:

    cfgo.values.webrequest_fixCSP = 'no'

[This page](https://developer.microsoft.com/en-us/microsoft-edge/platform/documentation/extensions/guides/debugging-extensions/) explains how to get there. Does this fix the issue?

**Comment by ccloli** [comment]
I tried, but seems that it doesn't work. Here are my steps:

1. I can't see the background page URL as an extension user, so I guest it would probably like Chrome extension so I tried `ms-browser-extension://EdgeExtension_JanBiniokTampermonkey_gz80c7jhhn2hw/background.html` and it opened;

2. Pressed F12 and switch to Console, then input `cfgo.values.webrequest_fixCSP`. I remembered it said its value is `yes`;

3. Then I tried `cfgo.values.webrequest_fixCSP = 'no'`, and checked `cfgo.values.webrequest_fixCSP` again, I remembered it said `no`;

4. Then I refreshed it again, but the page is still downloaded.

![image](https://cloud.githubusercontent.com/assets/8115912/18210261/5333ef5c-716a-11e6-907c-bb9c1cee9d35.png)

*****

BTW, I should screenshot step 2 and 3 but forgot, so I tried restarting Edge again to retry. But when I accessed to `background.html`, the value of `cfgo.values.webrequest_fixCSP` was `no`. So I thought it may cached the setting, then I removed Tampermonkey and install again. 

But after installed I can't open the popup page, and when I accessed to option page, it said an internal error happened, if restarted and happened again, press "Confirm" to report the problem in forum. I also can't access to background page.

So at this time I can't to check it again, and I'm not sure should I opened a new issue here. Because I run Windows 10 on a virtual machine, and I don't activate it, besides the virtual machine only have limited RAM, CPU resource and low HDD speed. With so many unstable reasons, I'm not sure it's a bug or problem of whom. So I'm just simply note them here.

@lychichem Please try `cfgo.values.webrequest_fixCSP = 'no'` again to check if the bug is still occurred, as I don't remember my steps clearly. Also try removing and installing Tampermonkey again, if you also have this problem, please consider open a new issue if needed.

**Comment by derjanb** [comment]
@ccloli OK, then the other way around. :) Please open the options page, press F12 (open developer tools), select the console tab and execute:

    sendMessage({ method: 'setOption', name: 'webrequest_fixCSP', value: 'no' }, function() {})

**Comment by derjanb** [comment]
I've added a temporary workaround... [4.2.5297](http://tampermonkey.net/changelog.php?version=4.2.5297&ext=gz80)

**Comment by ccloli** [comment]
Hi, I've already reinstalled Windows 10, so it's now in a clean environment. I tried the second option:
```
sendMessage({ method: 'setOption', name: 'webrequest_fixCSP', value: 'no' }, function() {})
```
And it works! Now Edge shows the page instead of asking to download. Thanks for your work. :-)

**Comment by lychichem** [comment]
@derjanb I'm sad to say that... but your fix do not always works on my machine... OK around 85% successful rate, sometime it still download the page, I just don't know why. I use tampermonkey edge 4.2.5294 now and UA is Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KMTML, like Gecko) Chrome/51.0.2704.79 Safari/537.36 Edge/14.14393

**Comment by derjanb** [comment]
I've published [4.2.5298](http://tampermonkey.net/changelog.php?version=4.2.5298&ext=gz80), but it might take some time until it is available. Can you please re-test with this version? Thanks.
